identity_stardust/src/state_metadata_document.rs

```diff
@@ -0,0 +1,326 @@
+// Copyright 2020-2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use identity_core::common::Object;
+use identity_did::did::CoreDID;
+use identity_did::document::CoreDocument;
+use identity_did::document::DocumentBuilder;
+use identity_did::verification::MethodRef;
+use serde::Deserialize;
+use serde::Serialize;
+
+use crate::did_or_placeholder::DIDOrPlaceholder;
+use crate::did_or_placeholder::PLACEHOLDER_DID;
+use crate::StardustDocument;
+
+/// The DID document as it is contained in the state metadata of an alias output.
+#[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
+pub struct StateMetadataDocument(CoreDocument<DIDOrPlaceholder>);
```

Comment:
This should be private and only used when `StardustDocument` is `pack`-ed into the state metadata.